server/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/daco/DaCoCopybookVisitor.java

```diff
@@ -0,0 +1,252 @@
+            .level(LEVEL_77)
+            .levelLocality(getLevelLocality(ctx.LEVEL_NUMBER_77()))
+            .variableNameAndLocality(extractNameAndLocality(ctx.entryName()))
+            .statementLocality(retrieveRangeLocality(ctx))
+            .global(!ctx.dataGlobalClause().isEmpty())
+            .picClauses(retrievePicTexts(ctx.dataPictureClause()))
+            .valueClauses(retrieveValues(ctx.dataValueClause()))
+            .usageClauses(retrieveUsageFormat(ctx.dataUsageClause()))
+            .occursClauses(retrieveOccursValues(ctx.dataOccursClause()))
+            .redefinesClauses(
+                    ctx.dataRedefinesClause().stream()
+                            .map(CobolParser.DataRedefinesClauseContext::dataName)
+                            .map(this::extractNameAndLocality)
+                            .collect(toList()))
+            .blankWhenZero(!ctx.dataBlankWhenZeroClause().isEmpty())
+            .signClause(!ctx.dataSignClause().isEmpty())
+            .build();
+    return addTreeNode(variableDefinitionNode, visitChildren(ctx));
+  }
+
+  @Override
+  public List<Node> visitDataDescriptionEntryFormat3(CobolParser.DataDescriptionEntryFormat3Context ctx) {
+    return ofNullable(ctx.dataValueClause())
+            .map(CobolParser.DataValueClauseContext::valueIsToken)
+            .map(
+                    valueToken ->
+                            addTreeNode(
+                                    VariableDefinitionNode.builder()
+                                            .level(LEVEL_88)
+                                            .levelLocality(getLevelLocality(ctx.LEVEL_NUMBER_88()))
+                                            .variableNameAndLocality(extractNameAndLocality(ctx.entryName()))
+                                            .statementLocality(retrieveRangeLocality(ctx))
+                                            .valueClauses(retrieveValues(ImmutableList.of(ctx.dataValueClause())))
+                                            .valueToken(retrieveValueToken(valueToken))
+                                            .build(),
+                                    visitChildren(ctx)))
+            .orElse(ImmutableList.of());
+  }
+
+  @Override
+  protected List<Node> defaultResult() {
+    return ImmutableList.of();
+  }
+
+  @Override
+  protected List<Node> aggregateResult(List<Node> aggregate, List<Node> nextResult) {
+    List<Node> result = new ArrayList<>(aggregate.size() + nextResult.size());
+    result.addAll(aggregate);
+    result.addAll(nextResult);
+    return result;
+  }
+
+  private int calculateLevel(int level) {
+    if (startingLevel != 0) {
+      if (firstCopybookLevel == 0) {
+        firstCopybookLevel = level;
+        return startingLevel;
+      }
+      return level - firstCopybookLevel + startingLevel;
+    }
+    return level;
+  }
+
+  private List<Node> addTreeNode(Node node, List<Node> children) {
```

Comment:
Should we have an interface that all visitor implements with some default implementation for common methods?

Probably not if, we need to override each of these.